portal-bridge/src/types/state/execution.rs

```diff
@@ -0,0 +1,168 @@
+use alloy_primitives::{keccak256, Address, Bytes, B256, U256};
+use alloy_rlp::Decodable;
+use anyhow::{ensure, Error};
+use eth_trie::{EthTrie, MemoryDB, Trie};
+use ethportal_api::types::state_trie::account_state::AccountState;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{BTreeSet, HashMap},
+    fs::File,
+    io::BufReader,
+    sync::Arc,
+};
+
+use crate::types::{era1::BlockTuple, state::blockreward::get_block_reward};
+
+#[derive(Debug, Serialize, Deserialize)]
+struct AllocBalance {
+    balance: U256,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct GenesisConfig {
+    alloc: HashMap<Address, AllocBalance>,
+    #[serde(rename = "stateRoot")]
```

Comment:
Imo using [this](https://github.com/ethereum/trin/blob/4d44d328c0c7e76a7619efa4283c33eb21f3ddfe/ethportal-api/src/types/state.rs#L9) would be better